packages/runtime/container-runtime/src/summarizerClientElection.ts

```diff
@@ -6,88 +6,120 @@
+        public readonly clientElection: IOrderedClientElection,
         private readonly maxOpsSinceLastSummary: number,
     ) {
         super();
-        this.summaryCollection.on("default", (op) => {
-            const opsSinceLastAckForClient = op.sequenceNumber - this.lastSummaryAckSeqForClient;
-            if (
-                opsSinceLastAckForClient > this.maxOpsSinceLastSummary
-                && !this.hasLoggedTelemetry
-                && this.electedClientId !== undefined
-            ) {
-                // Limit telemetry to only next client?
+        // On every inbound op, if enough ops pass without seeing a summary ack (per elected client),
+        // elect a new client and log to telemetry.
+        this.summaryCollection.on("default", ({ sequenceNumber }) => {
+            const electedClientId = this.electedClientId;
+            if (electedClientId === undefined) {
+                // Reset election if no elected client, but eligible clients are connected.
+                // This should be uncommon, but is possible if the initial state of the
+                // ordered client election contains an undefined client id or one not found
+                // in the quorum (the latter would already log an error).
+                if (this.clientElection.eligibleCount > 0) {
+                    this.clientElection.resetElectedClient(sequenceNumber);
+                }
+                return;
+            }
+            let electionSequenceNumber = this.clientElection.electionSequenceNumber;
+            const opsWithoutSummary = sequenceNumber - (this.lastSummaryAckSeqForClient ?? electionSequenceNumber);
+            if (opsWithoutSummary > this.maxOpsSinceLastSummary) {
+                // Log and elect a new summarizer client.
                 this.logger.sendErrorEvent({
                     eventName: "ElectedClientNotSummarizing",
-                    electedClientId: this.electedClientId,
+                    electedClientId,
                     lastSummaryAckSeqForClient: this.lastSummaryAckSeqForClient,
+                    electionSequenceNumber,
+                    nextElectedClientId: this.clientElection.peekNextElectedClient()?.clientId,
                 });
 
-                // In future we will change the elected client.
-                // this.orderedClients.incrementCurrentClient();
+                this.clientElection.incrementElectedClient(sequenceNumber);
 
-                this.hasLoggedTelemetry = true;
+                // Verify that we will not be reentrant. This should be reliable,
+                // since all of OrderedClientElection is synchronous.
+                electionSequenceNumber = this.clientElection.electionSequenceNumber;
+                if (sequenceNumber > (this.lastSummaryAckSeqForClient ?? electionSequenceNumber)) {
+                    this.logger.sendErrorEvent({
+                        eventName: "UnexpectedElectionSequenceNumber",
+                        sequenceNumber,
```

Comment:
>sequenceNumber [](http://example.com/codeflow?start=24&length=14)

this is too generic of a name, and is already logged at the contianer level

Reply:
Removed it, good call.